### PR TITLE
Added warning if saving >10GB on workbench. 

### LINKED ISSUE #26048: Project Saving: Warn if the saving over 10GB of data on workbench
Project saving will save all loaded workspaces, if those workspaces are large it may take some time.

If the sum total size of the workspaces to save is beyond 10GB (this should be defined in Mantid.,properties) then we should pop up a helpful are you sure box, warning them it may take some time and giving them options to continue or cancel.

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
User is now presented with a dialog box if the project that they are saving is over 10GB in size, telling them that the project may take a long time to save. 
The user can decide to continue with the save or cancel it. 

**To test:**
1. Create a project with >10GB of workspaces.
     a. Alternatively, change the `mantid.properties` file's `projectSaving.warningSize` value to a more reasonable number to reach. 
2. Attempt to save the project.
3. Check that clicking Cancel/x button does not save the project and clicking Yes does.
4. Clear the project and ensure that the dialog does not appear when saving a small project. 

Resolves #26048 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/project/project.py [hunk 1]
```diff
@@ -167,6 +185,16 @@ def _offer_save_message_box(self, parent):
         else:
             return QMessageBox.No
 
+    @staticmethod
+    def offer_large_size_confirmation():
```
Comment (Pasarus) [inline qt/python/mantidqt/project/project.py]: This probably needs to be a private method
```suggestion
    def _offer_large_size_confirmation():
```

File: qt/python/mantidqt/project/project.py [hunk 2]
```diff
@@ -106,12 +108,28 @@ def _save_file_dialog(self):
...
-        project_saver.save_project(file_name=self.last_project_location, workspace_to_save=workspaces_to_save,
-                                   plots_to_save=plots_to_save, interfaces_to_save=interfaces_to_save)
-        self.__saved = True
+        # Calculate the size of the workspaces in the project.
+        project_size = self._get_project_size(workspaces_to_save)
+        warning_size = int(ConfigService.getString("projectSaving.warningSize"))
+        # If a project is > the value in the properties file, question the user if they want to continue.
+        saving = True
+        if project_size > warning_size:
+            result = self.offer_large_size_confirmation()
```
Comment (Pasarus) [inline qt/python/mantidqt/project/project.py]:
```suggestion
            result = self._offer_large_size_confirmation()
```

File: qt/python/mantidqt/project/project.py [hunk 3]
```diff
@@ -13,11 +13,13 @@
 from qtpy.QtWidgets import QFileDialog, QMessageBox
 
 from mantid.api import AnalysisDataService, AnalysisDataServiceObserver
+from mantid.kernel import ConfigService
 from mantidqt.io import open_a_file_dialog
 from mantidqt.project.projectloader import ProjectLoader
 from mantidqt.project.projectsaver import ProjectSaver
 
 
+# noinspection PyTypeChecker
```
Comment (Pasarus) [inline qt/python/mantidqt/project/project.py]: Why are we turning off the PyTypeChecker?

File: qt/python/mantidqt/project/project.py [hunk 4]
```diff
@@ -106,12 +108,28 @@ def _save_file_dialog(self):
...
-        interfaces_to_save = self.interface_populating_function()
-        project_saver = ProjectSaver(self.project_file_ext)
-        project_saver.save_project(file_name=self.last_project_location, workspace_to_save=workspaces_to_save,
-                                   plots_to_save=plots_to_save, interfaces_to_save=interfaces_to_save)
-        self.__saved = True
+        # Calculate the size of the workspaces in the project.
+        project_size = self._get_project_size(workspaces_to_save)
+        warning_size = int(ConfigService.getString("projectSaving.warningSize"))
+        # If a project is > the value in the properties file, question the user if they want to continue.
+        saving = True
```
Comment (Pasarus) [inline qt/python/mantidqt/project/project.py]: I think this bool can be replaced with an if.

```python
result = None
if project_size > warning_size:
  result = self.offer_large_size_confirmation

if isinstance(result, None):
  #do saving stuff
```

File: qt/python/mantidqt/project/project.py [hunk 5]
```diff
@@ -106,12 +108,28 @@ def _save_file_dialog(self):
...
+                saving = False
+        if saving:
+            plots_to_save = self.plot_gfm.figs
+            interfaces_to_save = self.interface_populating_function()
+            project_saver = ProjectSaver(self.project_file_ext)
+            project_saver.save_project(file_name=self.last_project_location, workspace_to_save=workspaces_to_save,
+                                       plots_to_save=plots_to_save, interfaces_to_save=interfaces_to_save)
+            self.__saved = True
+
+    def _get_project_size(self, workspace_names):
```
Comment (Pasarus) [inline qt/python/mantidqt/project/project.py]:
```suggestion
    @staticmethod
    def _get_project_size(self, workspace_names):
```

**Review by Pasarus — CHANGES_REQUESTED** [review]
Overall this was implemented very well. I have a few suggestions as to the code structure and the _save class method is getting a little big but it's fine for its purpose.

File: qt/python/mantidqt/project/project.py (hunk 3 above)
Comment (ConorMFinn) [reply qt/python/mantidqt/project/project.py]: I have no idea, I don't think I put that in.

**Review by Pasarus — APPROVED** [review]
Looks good and functions well!

**Pull request merged**